src/lib/storage/value_segment.cpp

```diff
@@ -1,73 +1,94 @@
 #include "value_segment.hpp"
+#include <optional>
 
 #include "type_cast.hpp"
 #include "utils/assert.hpp"
 
 namespace opossum {
 
 template <typename T>
-ValueSegment<T>::ValueSegment(bool nullable) {
-  // Implementation goes here
-}
+ValueSegment<T>::ValueSegment(bool nullable) : _is_nullable(nullable) {}
 
 template <typename T>
 AllTypeVariant ValueSegment<T>::operator[](const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (is_null(chunk_offset)) {
+    return NULL_VALUE;
+  }
+  return get(chunk_offset);
 }
 
 template <typename T>
 bool ValueSegment<T>::is_null(const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return is_nullable() && null_values()[chunk_offset];
 }
 
 template <typename T>
 T ValueSegment<T>::get(const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  Assert(!is_null(chunk_offset), "Tried to .get a NULL value from a ValueSegment.");
+  return values()[chunk_offset];
```

Comment:
```suggestion
  return values().at(chunk_offset);
```
You could consider accessing an element of the vector with bounds checking.

Reply:
Let's make sure to stay consistent with bounds checking though, in other places we used explicit `Assert`ions, but I now tend towards using `at` everywhere too :)